### PR TITLE
SOL-71 [FE] Implement route protection

### PR DESCRIPTION
**Pull request by jasonchuasun**
## Issue Link
- Backlog [Link](https://framgiaph.backlog.com/view/SPH_SOLIDITY-71) 

## Target Page | Component | Function
- Add `/profile` and `/edit-campaign`

## Pre-condition
- test must be done for user not connected and is connected

## Steps | Commands
1. unconnected try visiting the private pages by manually typing the URL - `/profile` and `/edit-campaign`
2. while connected try visiting the private pages again
3. while on a private page, disconnect the wallet

## Expected Output
- not connected == redirect to home page
- connected == page is displayed
- while on private page and disconnected == redirect to home page

## Definition of Done
- route protection and page redirect should be working for connected/disconnected users

## Test View Points
- [ ] Add here

## Notes
- N/A

### General
- [ ] Does the code meet the requirements of the project?
- [ ] Is the code readable and maintainable?
- [ ] Are there any unnecessary or redundant lines of code?
- [ ] Are there any long methods or functions that should be refactored?
- [ ] Are there any performance issues?

### Functionality
- [ ] Are there any logical errors?
- [ ] Are edge cases handled properly?
- [ ] Are there any potential security vulnerabilities?

### Code Style
- [ ] Does the code adhere to the project's coding standards?
- [ ] Are variable and function names descriptive and meaningful?
- [ ] Is there proper use of comments and documentation?
- [ ] Is the code properly formatted? Are indentation, spacing, and formatting consistent?

### Security
- [ ] Is the website secure from common attacks?
- [ ] Are inputs properly validated and sanitized?
- [ ] Are sensitive data and credentials properly protected?

### Testing
- [ ] Are there sufficient test cases?
- [ ] Do the test cases cover all functionality?
- [ ] Are the test cases properly written and documented?

### Additional Comments
- [ ] Are there any other issues or concerns with the code?
- [ ] Are there any suggestions for improvement?
- [ ] Are there any positive aspects of the code that should be highlighted?

### CODE REVIEW

**Review by johnpaul-sun — DISMISSED** [review]
Approved ✅

Review Time: 15 mins
Comments: 0
IT: 0

**Review by arden-dave-sun — DISMISSED** [review]
Approved ✅
Review time: 10 mins
Comment: 0
Iteration: 0

**Review by johnpaul-sun — DISMISSED** [review]

**Review by arden-dave-sun — DISMISSED** [review]

**Review by arden-dave-sun — DISMISSED** [review]

**Comment by ipilikent** [comment]
## Approved PR
- Not connected wallet== The user can redirect to home page
- Connected Wallet == The page is displayed
- while on private page and disconnected wallet == The user can redirect to home page

## Proof
<details>

Sol- 71 Intregrate Route protection

</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=1622740366
-----
- Total Test time:  1 hr.
- Total Comments: 0
- Total Iterations: 0

**Review by ipilikent — APPROVED** [review]

File: fe/src/components/global/wallet-button.vue [hunk 1]
```diff
@@ -83,7 +85,14 @@ const connectWallet = async (): Promise<void> => {
...
   updateStatus(false, "");
-  toast.success("Wallet disconnected successfully!", { autoClose: 1500 });
+  useCookie("is-connected").value = null;
+
+  toast.success("Wallet disconnected successfully!", {
+    autoClose: 1500,
+  });
+  setTimeout(() => {
+    router.push("/");
+  }, 1500);
```
Comment (chevyB) [inline fe/src/components/global/wallet-button.vue:96]: this next line will not be executed and its useless right now:
`localStorage.removeItem("is-connected");`

**Review by chevyB — CHANGES_REQUESTED** [review]

File: fe/src/components/global/wallet-button.vue (hunk 1 above)
Comment (jasonchuasun) [reply fe/src/components/global/wallet-button.vue:96]: what do you mean by this sir? 
localstorage.removeItem('is-connected') still ran as per checking devtools.

**Review by johnpaul-sun — DISMISSED** [review]

**Review by chevyB — APPROVED** [review]

**Review by arden-dave-sun — APPROVED** [review]

**Pull request merged**